scarb/src/ops/publish.rs

```diff
@@ -51,10 +52,18 @@ pub fn publish(package_id: PackageId, opts: &PublishOpts, ws: &Workspace<'_>) ->
         .print(Status::new("Uploading", &dest_package_id.to_string()));
 
     ws.config().tokio_handle().block_on(async {
-        registry_client.publish(package, tarball).await
+        let upload = registry_client.publish(package, tarball).await;
+        match upload {
+            Ok(RegistryUpload::Success) => {
+                ws.config().ui().print(Status::new(
+                    "Published",
+                    format!("{}", &dest_package_id).as_str(),
+                ));
+                Ok(())
+            }
+            _ => upload.map(|_| ()),
```

Comment:
do I see correctly that you're ignoring any errors and you're not informing the user about the cause?

Reply:
Err are propagated, they are not consumed or modified in any way. You can see tests from registry_upload.rs (added i this PR)